core/src/main/java/org/ehcache/core/internal/service/ServiceLocator.java

```diff
@@ -426,7 +427,7 @@ public void loadDependenciesOf(Class<?> clazz) {
             throw new IllegalStateException("Service dependency declared by " + clazz.getName() +
                 " is not a Service: " + dependency.getName());
           }
-          dependencies.add(dependency);
+          dependencies.add((Class<? extends Service>) dependency);
```

Comment:
There is a check to see if the class is actually an instance of `Service`. So it doesn't make sense to type it as a `(Class<? extends Service>)`.

However, since the annotation is now typed, we probably can remove all these checks. But I've left that to another time.